### ISSUE TITLE
Remove IsOnOpenShift() from Kubevirt

### ISSUE THREAD

**Issue by iholder101** [kubevirt/kubevirt#13206]
**What happened**:
Our codebase uses the IsOnOpenShift() function.
https://github.com/kubevirt/kubevirt/blob/ced9fe1b5104daa431bed6b65ea690639fe3b79f/pkg/util/cluster/cluster.go#L22

**What you expected to happen**:
Kubevirt should be platform agnostic, and shouldn't container OpenShift specific code.
Any opinionated configuration needs to be done via HCO.

**Comment by iholder101** [comment]
> Hello, does that means that the openshift code sections of this repo needs to be migrated to the HCO repo?

The way I see it, Kubevirt should be customizable via some external parameters that would be handed by HCO whenever needed. There aren't many uses of `IsOnOpenShift()` in the production code, so I think it shouldn't be too complex to achieve.

Let me know if you want to work on this and you need some guidance :+1:

**Comment by damsien** [comment]
Yes I would like to work on it! And yes I would like to have some guidance as well 😅
